kernel/init.c

```diff
@@ -666,6 +667,7 @@ FUNC_NORETURN void z_cstart(void)
 	arch_smp_init();
 #endif
 	z_sys_init_run_level(INIT_LEVEL_PRE_KERNEL_2);
+	init_sys_clock_driver();
```

Comment:
Perhaps invert these 2 lines? 
clock priority is 0, so started as one of the very first of PRE_KERNEL_2

Reply:
yeah, we should not be far away from getting rid of PRE_KERNEL_2 anyways :)